src/extension.ts

```diff
@@ -19,7 +22,7 @@ var publicMPToken = "6a946c760957a81165973cc1ad5812ec";
 // your extension is activated the very first time the command is executed
 export async function activate(context: vscode.ExtensionContext) {
   // Setup our Telemetry Service
-  var telemetryService = new TelemetryService(publicMPToken);
+  var telemetryService = createTelemetryService(publicMPToken);
```

Comment:
What happened here? This seems like an unnecessary change, and moves us away from the provider architecture.

Reply:
This incorporates a singleton architecture which allows us to access services more fluidly between services

Reply:
Right, let's discuss tomorrow and walk through this a bit more.